### PR TITLE
Allow SProp default on

### PR DESCRIPTION
**Pull request by SkySkimmer**
Overlays: https://github.com/Karmaki/coq-dpdgraph/pull/67 https://github.com/LPCIC/coq-elpi/pull/83

### CODE REVIEW

**Comment by JasonGross** [comment]
Question: does SProp allow you to prove UIP for Logic.eq?  If yes, does it continue to do so when -indices-matter is enabled?

**Comment by ppedrot** [comment]
@JasonGross no, not in the current implementation.

**Comment by Zimmi48** [comment]
Is there some context on why the option was off by default initially and it has been decided to turn it on? Is it the result of WG discussions?

**Comment by gares** [comment]
It was discussed, indeed there is no reason to have it off.

**Comment by Zimmi48** [comment]
If there is absolutely no reason to have it off, then it could still be time to do this in 8.10 as well (if there is no risk associated to this) and then, maybe, the options and command-line could be simply removed... But this would need a more convincing story of why it was thought to be needed in the first place, and why this opinion changed.

**Comment by gares** [comment]
The reason it was there was for perf, but actually you pay the little cost anyway, so why bother.
For 8.10, I'm not the one to take the decision. The overlay for elpi is trivial, I had a chat with gaetan, if he is busy I can prepare it tomorrow.

**Comment by SkySkimmer** [comment]
>maybe, the options and command-line could be simply removed...

I think we want to keep at least one (probably the option) to allow people to control what theory they work with. In the same direction we may someday add options to forbid using inductive types (unimath), Prop, etc

**Comment by vbgl** [comment]
Adding the “-disallow-sprop” flag is a good idea for 8.10. Enabling SProp by default looks like a bigger change to do so late in the release cycle.

File: toplevel/usage.ml [hunk 1]
```diff
@@ -79,6 +79,7 @@ let print_usage_common co command =
 \n  -dump-glob f           dump globalizations in file f (to be used by coqdoc)\
 \n  -impredicative-set     set sort Set impredicative\
 \n  -allow-sprop           allow using the proof irrelevant SProp sort\
+\n  -disallow-sprop        forbid using the proof irrelevant SProp sort\
```
Comment (SkySkimmer) [inline toplevel/usage.ml]: This is just a doc fix, -disallow-sprop has worked since the beginning.

**Review by Zimmi48 — APPROVED** [review]
This should get a changelog entry though.

**Comment by Zimmi48** [comment]
Note that the command-line flag could also be removed in favor of `-set "Allow SProp=true"` if it is not that useful.

File: doc/changelog/01-kernel/10811-sprop-default-on.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- Using ``SProp`` is now allowed by default, without needing to pass
+  ``-allow-sprop`` or use :opt:`Allow StrictProp` (`#10811
```
Comment (Zimmi48) [inline doc/changelog/01-kernel/10811-sprop-default-on.rst]:
```suggestion
  ``-allow-sprop`` or use :flag:`Allow StrictProp` (`#10811
```

**Comment by SkySkimmer** [comment]
Is this waiting for anything? Who shall assign?

**Review by ejgallego — APPROVED** [review]
Looks good, I guess 8.11 RM may be the ones to take care of this.

**Review by ppedrot — APPROVED** [review]
Good for me, let's merge.

**Pull request merged**